opensc-notify: register a desktop menu entry

frankmorgner opened the pull request:
- fixes showing notifications in gnome-shell via gio2
- removes gdbus interface for notifications
- fixes #1186

##### Checklist

- [x] Tested with the following card: sc-hsm
	- [x] tested PKCS#11

frankmorgner commented:
@Jakuje could you check whether this fixes your problem?

Jakuje commented:
No. With this patch, it fails the show the notification altogether with the following error even if I have a display:
```
(process:18872): GLib-GIO-CRITICAL **: g_application_send_notification: assertion 'G_IS_APPLICATION (application)' failed
```
But I was testing just from the local build without installation (though the current master work with this setup properly).

frankmorgner commented:
I forgot to commit a change that was present locally. Sorry, for the inconvenience.

Jakuje commented:
This does not throw error messages, but does not show the notification either in GNOME as it was showing before. Does it require to install the desktop entry somewhere?

frankmorgner commented:
Yes, gnome shell requires you to put it into one of `$XDG_DATA_DIRS`'s `applications` directories (e.g. putting `org.opensc-project.opensc-notify.desktop` into `~/.local/share/applications/` will do)

Jakuje commented on src/tools/Makefile.am:
```diff
@@ -178,5 +179,11 @@ gids_tool_SOURCES += versioninfo-tools.rc
+applicationsdir = $(datadir)/applications
+applications_DATA = org.opensc-project.opensc-notify.desktop
+
+org.opensc-project.opensc-notify.desktop: org.opensc-project.opensc-notify.desktop
```
This should say (adding `.in` to the end):
```
org.opensc-project.opensc-notify.desktop: org.opensc-project.opensc-notify.desktop.in
```

frankmorgner replied:
done, thanks

Jakuje commented:
For some reason, I don't have `~/.local/share` in `$XDG_DATA_DIRS`. But even after putting it in into `/usr/share/applications/` and after running `desktop-file-validate` with this desktop file I do not see any notification nor any error. Hard to say what could go wrong.

Looking into the configure log, it looks like everything is defined as expected. Not sure what else I could check or if the debug log would be useful.

frankmorgner commented:
Strange... have you tried a different desktop environment? AFAIK, gnome shell is the only one that requires you to have a desktop entry.

Jakuje commented:
I don't have any other DE installed and set up to try at this moment.

I tried to intercept the notification using DBus, but I did not see them either according to the following post but nothing happens (for example using `pkcs11-tool`):

https://askubuntu.com/questions/105566/is-there-a-way-to-view-notification-history/352125#352125

On the other hand, I can see the same failures
for other tools using `libopensc`, for example if I try  `tests/pintest`, which is not improvement either (before applying this patch, it was showing the notification properly even from here).
```
(process:20090): GLib-GIO-CRITICAL **: g_application_send_notification: assertion 'G_IS_APPLICATION (application)' failed
```
There should be some check, that the notify is initialized, before trying to show notification, should not it?

I even tried to run the example notification program using GIO2 from archlinux, but still no notification:

https://wiki.archlinux.org/index.php/Desktop_notifications

Though I am able to get the notifications using `libnofity` example program and I am getting the notifications from other programs on my desktop. I am probably out of ideas where to search or what to try so at least these ideas so far.

frankmorgner commented:
I had the same problem for quite some time, that's why I've implemented the `gdbus` workaround. But as explained above, I solved the problem with a desktop file for gnome-shell. I hope that the default install location I've used will work on other distributions, too...

I've added the check on whether the notifications are initialized.

frankmorgner commented:
I've noticed that I've got the following in my shell startup configuration:
```
export XDG_DATA_DIRS=$XDG_DATA_DIRS:$HOME/.local/share
```
Setting this may let xdg find your desktop file.
